### PR TITLE
Add support for downloading files to the storage client.

### PR DESCRIPTION
**Pull request by ahoying**
This is required for the CAI integration work.

Thanks for opening a Pull Request!

Here's a handy checklist to ensure your PR goes smoothly.

- [x] I signed Google's [Contributor License Agreement](https://opensource.google.com/docs/cla/)
- [x] My code conforms to Google's [python style](https://google.github.io/styleguide/pyguide.html).
- [x] My PR at a minimum doesn't decrease unit-test coverage (if applicable).
- [x] My PR has been functionally tested.
- [x] All of the [unit-tests](https://forsetisecurity.org/docs/latest/develop/dev/testing.html) still pass.
- [x] Running `pylint --rcfile=pylintrc` passes.

These guidelines and more can be found in our [contributing guidelines](https://github.com/GoogleCloudPlatform/forseti-security/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: google/cloud/forseti/common/gcp_api/storage.py [hunk 1]
```diff
@@ -312,6 +312,29 @@ def download(self, bucket, object_name):
             out_stream.close()
         return file_content
 
+    def download_to_file(self, bucket, object_name, output_file):
+        """Download an object from a bucket.
+         Args:
+            bucket (str): The name of the bucket to read from.
+            object_name (str): The name of the object to read.
+            output_file (file): The file object to write the data to.
+         Returns:
```
Comment (blueandgold) [inline google/cloud/forseti/common/gcp_api/storage.py]: nit: newline is needed here

File: google/cloud/forseti/common/gcp_api/storage.py [hunk 2]
```diff
@@ -312,6 +312,29 @@ def download(self, bucket, object_name):
             out_stream.close()
         return file_content
 
+    def download_to_file(self, bucket, object_name, output_file):
+        """Download an object from a bucket.
+         Args:
```
Comment (blueandgold) [inline google/cloud/forseti/common/gcp_api/storage.py]: nit: newline is needed here

File: google/cloud/forseti/common/gcp_api/storage.py [hunk 3]
```diff
@@ -419,6 +442,28 @@ def get_text_file(self, full_bucket_path):
             LOGGER.exception('Unable to download file.')
             raise
 
+    def download_file(self, full_bucket_path, output_file):
+        """Downloads a copy of a file from GCS.
+         Args:
```
Comment (blueandgold) [inline google/cloud/forseti/common/gcp_api/storage.py]: nit: newline is needed here

File: google/cloud/forseti/common/gcp_api/storage.py [hunk 4]
```diff
@@ -419,6 +442,28 @@ def get_text_file(self, full_bucket_path):
...
+            full_bucket_path (str): The full path of the bucket object.
+            output_file (file): The file object to write the data to.
+         Returns:
```
Comment (blueandgold) [inline google/cloud/forseti/common/gcp_api/storage.py]: nit: newline is needed here

File: google/cloud/forseti/common/gcp_api/storage.py [hunk 5]
```diff
@@ -419,6 +442,28 @@ def get_text_file(self, full_bucket_path):
...
+            int: Total size in bytes of file.
+         Raises:
```
Comment (blueandgold) [inline google/cloud/forseti/common/gcp_api/storage.py]: nit: newline is needed here

File: google/cloud/forseti/common/util/file_loader.py [hunk 6]
```diff
@@ -43,6 +44,29 @@ def read_and_parse_file(file_path):
     return _read_file_from_local(file_path)
 
 
+def copy_file_from_gcs(file_path, output_path=None, storage_client=None):
+    """Copy file from GCS to local file.
+     Args:
```
Comment (blueandgold) [inline google/cloud/forseti/common/util/file_loader.py]: nit: newline is needed here

File: google/cloud/forseti/common/util/file_loader.py [hunk 7]
```diff
@@ -43,6 +44,29 @@ def read_and_parse_file(file_path):
...
+        file_path (str): The full GCS path to the file.
+        output_path (str): The local file to copy to, if not set creates a
+            temporary file.
+        storage_client (storage.StorageClient): The Storage API Client to use
+            for downloading the file using the API.
+     Returns:
```
Comment (blueandgold) [inline google/cloud/forseti/common/util/file_loader.py]: nit: newline is needed here

File: google/cloud/forseti/common/gcp_api/storage.py [hunk 8]
```diff
@@ -419,6 +442,28 @@ def get_text_file(self, full_bucket_path):
...
+            HttpError: HttpError is raised if the call to the GCP storage API
+                fails
+        """
+        bucket, object_name = get_bucket_and_path_from(full_bucket_path)
+        try:
+            file_size = self.repository.objects.download_to_file(
```
Comment (blueandgold) [inline google/cloud/forseti/common/gcp_api/storage.py]: nit: the `download_to_file()` method seems a bit confusing with the `download_file()`.  Perhaps, tweak `download_file()` to just `download()`?

Comment (ahoying) [reply google/cloud/forseti/common/gcp_api/storage.py]: Done.

**Pull request merged**